hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -1087,22 +1083,29 @@ extends Importer:
               
           val ctxWithFields = ctx
             .withMembers(
-              fields.fold(Nil)(_.collect:
-                case f: TermDefinition => f.sym.nme -> f.sym // class fields
-              ),
+              fields.foldRight(Nil):
```

Comment:
Use `fields.iterator.collect` instead?

Reply:
`fields.collect` will allocate a new list in memory, where every element is wrapped in a separate cons cell. Using an iterator means we'll only allocate an iterator object (which uses mutation internally).